File: Problem7/euler.py
def is_prime(n):
    if n == 2 or n == 3: return True
    if n < 2 or n%2 == 0: return False
    if n < 9: return True
    if n%3 == 0: return False
    r = int(n**0.5)
    f = 5
    while f <= r:
        if n%f == 0: return False
        if n%(f+2) == 0: return False
        f +=6
    return True 

list = []
for i in range(104744):
    if is_prime(i):
        list.append(i)

for i in range(int(raw_input())):
    print list[int(raw_input())-1]

File: Problem48/euler.cpp
////////////// 2 cases time out

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;

/*
import math

sum1 = 0
n = int(raw_input())
temp1 = 10**11
for i in range(1, n+1):
    temp = pow(i, i, temp1)
    temp = temp % temp1
    sum1 = sum1 + temp
str1 = str(sum1)
print int(str1[len(str1)-10:len(str1)])
*/

typedef unsigned long long int ull;
ull temp1 = pow(10, 10);    

ull powergt(ull a, ull b, ull c)
{
    ull product = 1;
    for(ull i =0; i < b; i++)
    {
        product *= a;
        if (product > temp1)
            product %= temp1;
    }
 return product;   
}

int main() {
    ull sum1 = 0, n, i, temp;
    cin>>n;
    for(i = 1; i < n+1; i++)
    {
        temp = powergt(i, i, temp1);
        temp = temp % temp1;
        sum1 = sum1 + temp;
    }
    cout<<(sum1 % temp1) ;
    return 0;
}
